packages/contracts/test/test-utils/uups-upgradeable.ts

```diff
@@ -2,27 +2,23 @@ import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
 import {expect} from 'chai';
 import {Contract, ContractFactory, errors} from 'ethers';
 import {upgrades} from 'hardhat';
-import {DAO} from '../../typechain';
+import {DAO, PluginRepo} from '../../typechain';
 import {readImplementationValueFromSlot} from '../../utils/storage';
+import {IMPLICIT_INITIAL_PROTOCOL_VERSION} from './protocol-version';
 
-// Deploys and upgrades a contract that is managed by a DAO
-export async function ozUpgradeCheckManagedContract(
+// Deploys a proxy and a new implementation from the same factory and checks that the upgrade works.
+export async function upgradeToSelfCheck(
```

Comment:
i see why you use self and other in the function name,
but perhaps only mentioning "upgrade" might cause confusion as it will upgrade something only....

what about:
`export async function deployAndUpgradeSelfCheck( ... )
`
`export async function deployAndUpgradeFromToCheck( ... )
`

another note:
I am wondering if we need 2 functions here?
we could have 1 function that can do all such as:
`
export async function deployAndUpgradeFromToCheck( 
    deployer: SignerWithAddress,
  upgrader: SignerWithAddress,
  initArgs: any,
  initializerName: string,
  upgradePermissionId: string,
  from: ContractFactory,
  to?: ContractFactory,
  managingDao?: DAO | PluginRepo
 )

`
WDYT?

Reply:
> what about:
export async function deployAndUpgradeSelfCheck( ... ) 
export async function deployAndUpgradeFromToCheck( ... ) 

sounds good

> I am wondering if we need 2 functions here?

In one case, we need to do different checks so I would keep them separate.

Reply:
Renamed in 945ddf39b92addab176866440f51db60abc36e72